torchdata/dataloader2/reading_service.py

```diff
@@ -94,6 +94,12 @@ def finalize_iteration(self) -> None:
         """
         pass
 
+    def __del__(self):
+        try:
+            self.finalize()
+        except AttributeError:
+            pass
+
```

Comment:
Change 2

Reply:
Just a comment - a little bit surprised we need this given `ProtoMultiRS`'s `finalize` seems to be catching every possible `AttributeError`? Maybe the issue is `Distributed`? Thoughts?

Reply:
`ProtoMPRS` doesn't handle all of `AttributeError` like the `self._worker_processes` from 
https://github.com/pytorch/data/blob/98222ad72ee7a29e676646e6b3f9173576410320/torchdata/dataloader2/reading_service.py#L345

Technical speaking, I should remove those try-except clauses in `finalize` to simplify the codebase